source/includes/inkling-reference/_schemas.html.md

```diff
@@ -1,265 +1,480 @@
-These range expressions specify sets of values in 
-which each value is explicitly listed.
+A value list range expression must conform to the signed or unsigned attribute
+of the specified type (if there is such an attribute). 
+
+###### Schema Matching
+
+Inkling uses schemas to understand and interpret the data format of streams.  The Inkling compiler performs schema match checking and will report errors if schemas which are expected to match do not. 
 
 ```inkling--code
-# Example: Range expression, colon range type
-schema Schema3
-  Int64  { 0:5:100 }   x,   # start:step:stop, step= 5,0..100
-  Int64  { 0:100 }     y    # start:stop, step= 1, 0..100
+schema GameState
+  Luminance(84, 336) pixels
+end
+ 
+concept keep_paddle_under_ball 
+  is classifier
+  predicts (PlayerMove)
+  follows ball_location, 
+          input(GameState)           # <--- (1) MATCH 
+end
+ 
+concept high_score 
+  is classifier
+  predicts (PlayerMove)
+  follows keep_paddle_under_ball, 
+          input(GameState)           # <--- (1) MATCH 
+  feeds output
+end
+ 
+concept ball_location 
+  is estimator
+  predicts (Matrix(UInt32, 1, 2) location)
+  follows 
+    input(Luminance(84, 336) pixels) # <--- (2) MATCH
+end
+ 
+concept ball_X_location is estimator
+  predicts (Uint32 X_location)
+  follows 
+    input(Luminance(84, 330) pixels) # <--- (3) NO MATCH
 end
 ```
 
-* **Colon range expression**
+Matching for schemas is both structural and name based. Field names matter.
+Types matter.
+
+* Two references by name to the same schema match because a schema matches itself. 
+* Two references to different schema names match if both schemas define the same list of field types in the same order with the same names. For structured types whose declaration includes size, the sizes must be equal.
+* Two anonymous schemas match if both define the same field types in the same order with the same names. 
+* A schema referenced by name matches an anonymous schema if both define the same field types in the same order with the same names. 
+
+The example to the right shows successful schema matching and failed schema matching. 
+
+* (1) shows a valid schema match by name. 
+* (2) shows a valid match by field type, field name, and size. 
+* (3) shows a failed match (size is not equal). 
+
+In this example we are matching against the schema associated with `input`.
+Every Inkling program has the predefined stream `input` available to it. Since there is a single stream associated with the keyword `input`, there can be only one definition of the data format of that stream. Once defined, the data format of a stream cannot change dynamically.  If that schema changes inkling compiler will flag that change as an error. Thus any reference to the `input` stream must have a schema that matches all other schemas used with the `input` stream. 
 
-Colon range expressions specify values for the start, 
-the step, and the stop. 
+###### Constraint Compatibility in the Lesson Configure Clause
```

Comment:
Does this section belong under "Types" or under "Schemas" in the outline?

Reply:
Constraint Compatibility is also a type relationship so I think it belongs under Types (which is under Schemas).